### ISSUE TITLE
Execution plan: architecture ledger, hybrid storage, novelty field, learned-Helix validation

### ISSUE THREAD

**Issue by euphoricdoom** [euphoricdoom/project-512d#1]
# Execution Plan: Project 512D Stabilization + Next Build

This issue tracks the next repo execution pass after the Phase 1 Helix AC/DC gate passed.

## Confirmed foundation

Phase 1 Helix validation is complete:

```text
1 kernels:   copy 100%, parity 100%, adding 100%, forgetting 0.0000
4 kernels:   copy 100%, parity 100%, adding 100%, forgetting 0.0000
16 kernels:  copy 100%, parity 100%, adding 100%, forgetting 0.0000
64 kernels:  copy 100%, parity 100%, adding 100%, forgetting 0.0000
```

Important caveat to preserve in all claims: this pass uses explicit task-aware temporal channels in Helix:

- compact copy memory slots
- explicit parity channel
- marker-aware adding accumulator

That is the validated baseline. The next research step is replacing or matching those channels with learned gates.

---

## Phase A — Truth and repo stabilization

Goal: make the repo self-explaining before adding more architecture.

### Add docs

- [ ] `docs/ARCHITECTURE_LEDGER.md`
- [ ] `docs/CLAIMS_AND_LIMITS.md`
- [ ] `docs/DEPENDENCY_TIERS.md`
- [ ] `docs/BACKBURNER.md`

### Clean up known inconsistencies

- [ ] README still says `v0.2`; update to current research/product status.
- [ ] README says `No PyTorch`, but learned-gate modules import torch. Clarify: core path is NumPy/no PyTorch; learned-gate research path has optional PyTorch.
- [ ] `requirements.txt` does not include torch, but `system/adaptive_helix.py`, `system/learned_gates.py`, `system/task_embedder.py`, and `system/gate_generator.py` require it.
- [ ] `docs/LATTICE_ARCHITECTURE.md` lists `bridge_threshold=0.15`, while `LatticeConfig` code currently defaults to `0.40`. Pick one and update docs/code.
- [ ] `__main__.py` experiment registry appears stale relative to current experiment files and docs.

### Add tooling

- [ ] `tools/repo_census.py`: generate a repo inventory covering modules, docs, tests, experiments, outputs, entrypoints, TODO/FIXME markers, and generated artifacts.

Acceptance:

- [ ] `python -m pytest -q` passes.
- [ ] No model behavior changes in Phase A unless explicitly required for doc/code mismatch.
- [ ] New docs clearly mark implemented / validated / experimental / planned / parked.

---

## Phase B — Hybrid readout storage

Goal: solve the coming storage scaling problem for task-specific heads.

Core strategy:

```text
Task head
  -> low-rank factorization
  -> sparse pruning
  -> int8 quantization
  -> compressed disk storage
  -> LRU cache
```

### B1: Minimal production-safe storage

- [ ] `storage/__init__.py`
- [ ] `storage/quantized_readouts.py`
- [ ] `storage/disk_cache.py`
- [ ] `tests/test_quantized_readouts.py`
- [ ] `tests/test_disk_cache.py`

Initial target:

- store task heads as int8 + scale
- load on demand from disk
- keep hot heads in bounded LRU cache
- verify prediction drift is below tolerance

### B2: Full hybrid compression

- [ ] `storage/low_rank.py`
- [ ] `storage/sparse.py`
- [ ] `storage/compressed_readouts.py`
- [ ] `benchmarks/storage_benchmark.py`

Acceptance:

- [ ] round-trip load/save preserves shape and metadata
- [ ] drift test reports max/mean prediction error
- [ ] storage size measured before/after compression
- [ ] cache hit/miss behavior tested

---

## Phase C — Novelty memory field

Goal: decide when Arbiter should reuse, shadow-test, spawn, or ignore a memory/readout node.

Core law:

```text
Novelty decides growth.
Resonance decides routing.
Stability blocks noise.
```

### Add

- [ ] `memory/__init__.py`
- [ ] `memory/novelty_field.py`
- [ ] `tests/test_novelty_field.py`
- [ ] `docs/NOVELTY_MEMORY_FIELD.md`

Expected routing decisions:

- `reuse`: strong resonance with existing node
- `shadow`: similar enough to test candidates before committing
- `spawn`: novel and stable enough to create new node
- `ignore_noise`: novel but unstable/noisy

Acceptance:

- [ ] first input spawns
- [ ] near input reuses
- [ ] medium input shadows
- [ ] far stable input spawns
- [ ] far unstable input ignores
- [ ] `top_k` is sorted by resonance
- [ ] deterministic IDs and stable serialization hooks

---

## Phase D — Learned-Helix validation

Goal: compare the explicit Helix baseline against learned/adaptive gates.

### Run matrix

```bash
python experiments/run_classic_benchmarks.py --kernels 1,4,16,64 --temporal-adapter none --epochs 30 --samples 512 --steps 3 --batch-size 32 --projection-from 1
python experiments/run_classic_benchmarks.py --kernels 1,4,16,64 --temporal-adapter helix --epochs 30 --samples 512 --steps 3 --batch-size 32 --projection-from 1
python experiments/run_classic_benchmarks.py --kernels 1,4,16,64 --temporal-adapter learned-helix --epochs 30 --samples 512 --steps 3 --batch-size 32 --projection-from 1
```

### Add report

- [ ] `docs/LEARNED_HELIX_COMPARISON.md`
- [ ] adapter-mode comparison table
- [ ] accuracy, forgetting, runtime, and caveats

Acceptance:

- [ ] explicit Helix remains the validated baseline
- [ ] learned-Helix result is reported honestly whether it passes or fails

---

## Phase E — Baseline comparison for paper credibility

Goal: provide a standard recurrent baseline.

- [ ] `baselines/vanilla_lstm.py`
- [ ] `experiments/run_lstm_baseline.py`
- [ ] `docs/LSTM_BASELINE_RESULTS.md`

Acceptance:

- [ ] same task generator as classic benchmark runner
- [ ] same train/test splits where practical
- [ ] report forgetting and final accuracy per task
- [ ] do not overclaim; compare against the implemented baseline only

---

## Priority order

1. Phase A docs + truth cleanup
2. Phase B1 disk-backed quantized storage
3. Phase C novelty memory field
4. Phase D learned-Helix comparison
5. Phase E LSTM baseline

## Notes

Do not add another large architecture layer until storage and novelty routing exist. The current repo already has enough architecture; the next wins come from durability, proof, and scale discipline.
